angular_gallery_section/lib/gallery_docs_extraction.dart

```diff
@@ -81,59 +119,123 @@ class GalleryDocumentaionExtraction
-    }
+/// A visitor that extracts a [PropertyInfo] for every @Input and @Output
+/// property.
+///
+/// Only passes [_filePath] through the the indivudual extractor each
+/// [PropertyInfo].
+class _AllMemberDocsExtraction
+    extends SimpleAstVisitor<Iterable<PropertyInfo>> {
+  final _MemberDocExtraction _propertyVisitor;
 
-    // Handle /* */-style comments
-    String comment = commentNode.tokens.single.lexeme;
-    Match match = _multiLineCommentStartEnd.firstMatch(comment);
-    if (match != null) {
-      comment = match[1];
-      var sb = new StringBuffer();
-      List<String> lines = comment.split('\n');
-      for (int index = 0; index < lines.length; index++) {
-        String line = lines[index].trimRight();
-        if (index == 0) {
-          sb.write(line); // Add the first line unprocessed.
-          continue;
-        }
-        sb.write('\n');
-        match = _multiLineCommentLineStart.firstMatch(line);
-        if (match != null) {
-          sb.write(match[1]);
-        } else {
-          sb.write(line);
-        }
-      }
-      return sb.toString().trim();
-    }
-    throw new ArgumentError('Invalid comment $comment');
+  _AllMemberDocsExtraction(_filePath)
+      : this._propertyVisitor = new _MemberDocExtraction(_filePath);
+
+  @override
+  visitConstructorDeclaration(ConstructorDeclaration node) =>
+      const Iterable.empty();
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) =>
+      [node.accept(_propertyVisitor)];
+
+  @override
+  visitFieldDeclaration(FieldDeclaration node) =>
+      node.fields.variables.map((field) => field.accept(_propertyVisitor));
+}
+
+/// A visitor that extracts comments and the name of a class field or method as
+/// a [PropertyInfo].
+///
+/// Only uses [_filePath] to store in the returned [PropertyInfo] to document
+/// the file it was extracted from.
+class _MemberDocExtraction extends SimpleAstVisitor<PropertyInfo> {
+  final String _filePath;
+
+  _MemberDocExtraction(this._filePath);
+
+  @override
+  visitMethodDeclaration(MethodDeclaration node) => extractProperty(node);
+
+  @override
+  visitVariableDeclaration(VariableDeclaration node) => extractProperty(node);
+
+  /// Extracts information for documenentation from a [MethodDeclaration] or
+  /// [VariableDeclaration].
+  PropertyInfo extractProperty(Declaration node) {
+    return new PropertyInfo()
+      ..name = (node as dynamic /* MethodDeclaration | VariableDeclaration */)
+          .name
+          .name
+      ..comment = g3docMarkdownToHtml(parseComment(node.documentationComment))
+      ..classPath = _filePath;
   }
 }
 
-/// Information used for documenation in the gallery.
-class DocumentationInfo {
-  /// The path in which this component was found.
-  String name;
-  String type;
-  String selector;
-  String comment;
+final RegExp _singleLineCommentStart = new RegExp(r'^///? ?(.*)');
+final RegExp _multiLineCommentStartEnd =
+    new RegExp(r'^/\*\*? ?([\s\S]*)\*/$', multiLine: true);
```

Comment:
You might want to make this sub-regexp part lazy:

- `([\s\S]*)` --> `([\s\S]*?)`

(where _e_\*? is the lazy version of _e_\*).

Reply:
Thanks, I'll investigate the difference.